contracts/StakingPoolDAO.sol

```diff
@@ -164,4 +169,72 @@ contract StakingPoolDAO is StakingPool, ERC20Snapshot, Controlled {
   function() external payable {
     //
   }
+
+  enum GSNErrorCodes {
+    FUNCTION_NOT_AVAILABLE,
+    HAS_ETH_BALANCE,
+    GAS_PRICE,
+    TRX_TOO_SOON,
+    ALREADY_VOTED,
+    NO_TOKEN_BALANCE
+  }
+
+  bytes4 constant VOTE_SIGNATURE = bytes4(keccak256("vote(uint256,bool)"));
+
+  function acceptRelayedCall(
+      address relay,
+      address from,
+      bytes calldata encodedFunction,
+      uint256 transactionFee,
+      uint256 gasPrice,
+      uint256 gasLimit,
+      uint256 nonce,
+      bytes calldata approvalData,
+      uint256 maxPossibleCharge
+  ) external view returns (uint256, bytes memory) {
+
+    bytes memory abiEncodedFunc = encodedFunction; // Call data elements cannot be accessed directly
+    bytes4 functionSignature;
+    uint proposalId;
+
+    assembly {
+      functionSignature := mload(add(abiEncodedFunc, add(0x20, 0)))
+      proposalId := mload(add(abiEncodedFunc, 36))
+    }
+
+    return _evaluateConditions(from, functionSignature, proposalId, gasPrice);
+  }
+
+  function _evaluateConditions(
+    address _from,
+    bytes4 _functionSignature,
+    uint _proposalId,
+    uint _gasPrice
+  ) internal view returns (uint256, bytes memory) {
+    if(_functionSignature != VOTE_SIGNATURE) return _rejectRelayedCall(uint256(GSNErrorCodes.FUNCTION_NOT_AVAILABLE));
+
+    Proposal storage proposal = proposals[_proposalId];
+
+    if(balanceOfAt(_from, proposal.snapshotId) == 0) return _rejectRelayedCall(uint256(GSNErrorCodes.NO_TOKEN_BALANCE));
+
+    /* ?
+    if(from.balance > 600000 * gasPrice) return _rejectRelayedCall(uint256(GSNErrorCodes.HAS_ETH_BALANCE));
+    */
+
+    if(_gasPrice > 20000000000) return _rejectRelayedCall(uint256(GSNErrorCodes.GAS_PRICE)); // 20 gwei
+
+    if((lastActivity[_from] + 15 minutes) > block.timestamp) return _rejectRelayedCall(uint256(GSNErrorCodes.TRX_TOO_SOON));
```

Comment:
Do we still want this? What would be the normal rate at which users make transactions?

Reply:
I'm not sure. 
This would be in cases where a single user will vote in more than one proposal. Might not be necessary though, because we are interested in having people voting, so limiting it to 1 vote each 15 minutes does not help.